misk-redis/src/testFixtures/kotlin/misk/redis/testing/FakeRedis.kt

```diff
@@ -411,4 +427,113 @@ class FakeRedis @Inject constructor(
   override fun publish(channel: String, message: String) {
     throw NotImplementedError("Fake client not implemented for this operation")
   }
+
+  private fun zadd(
+    key: String,
+    score: Double,
+    member: ByteString,
+    options: Set<Redis.ZAddOptions>,
+  ): Long {
+    verifyZAddOptions(options)
+    var newFieldCount = 0L
+    var elementsChanged = 0L
+    val trackChange = options.contains(CH)
+
+    if (!sortedSetKeyValueStore.containsKey(key)) {
+      sortedSetKeyValueStore[key] = Value(data = hashMapOf(), expiryInstant = Instant.MAX)
+    }
+    val sortedSet = sortedSetKeyValueStore[key]!!.data
+    val exists = sortedSet[member] != null
+
+    if (shouldUpdateScore(sortedSet[member], score, exists, options)) {
+      sortedSet[member] = score
+      elementsChanged++
+      if (!exists) newFieldCount++
+    }
+
+    if (trackChange) return elementsChanged
+    return newFieldCount
+  }
+
+  /**
+   * If [exists], the [currentScore] will be present. If not, [currentScore] will be null
+   */
+  private fun shouldUpdateScore(
+    currentScore: Double?,
+    score: Double,
+    exists: Boolean,
+    zaddOptions: Set<Redis.ZAddOptions>
+  ) : Boolean {
+    val options = zaddOptions.filter { it != CH }
+    // default without any options
+    if (options.isEmpty()) return true
+
+    // all valid single options.
+    if ((options.size == 1)
+        && (((options[0] == XX) && exists)
+        || ((options[0] == NX) && !exists)
+        || ((options[0] == LT) && ((exists && score < currentScore!!) || !exists))
+        || ((options[0] == GT) && ((exists && score > currentScore!!) || !exists)))
+    ) return true
+
+    // valid option combos
+    // only two valid combos of two option are possible.
+    // LT XX and GT XX
+
+    // LT XX
+    // for existing ones, the score should be less than the existing scores.
+    // XX will prevent adding new ones.
+    if (options.contains(LT) && options.contains(XX) && exists
+      && score < currentScore!!) return true
+
+    // GT XX
+    // for existing ones, the score should be more than the existing scores.
+    // XX will prevent adding new ones.
+    if (options.contains(GT) && options.contains(XX) && exists
+      && score > currentScore!!) return true
+
+    return false
```

Comment:
You can simplify this logic a lot if you exit early for the `XX` and `NX` modifiers.
```suggestion
    if ((options.contains(XX) && !exists) || (options.contains(NX) && exists)) {
      return false
    }

    if (options.contains(LT)) {
      return !exists || score < currentScore!!
    }

    if (options.contains(GT)) {
      return !exists || score > currentScore!!
    }
   
    return true
```

Reply:
Spoke offline. I think the existing one is easier to understand if you read the comments on each option in the docs. So keeping it.